File: backend/notifications/src/main/java/com/oxygensend/notifications/domain/DomainFactory.java
package com.oxygensend.notifications.domain;

import com.oxygensend.notifications.context.dto.MailDto;
import com.oxygensend.notifications.context.dto.SmsDto;

public class DomainFactory {
    private DomainFactory() {
    }

    public static Mail from(MailDto mailDto) {
        return new Mail(mailDto.subject(), mailDto.body());
    }

    public static Sms from(SmsDto smsDto) {
        return new Sms(smsDto.body());

    }

    public static Email from(String email) {
        return new Email(email);
    }

}

File: backend/notifications/src/main/kotlin/com/oxygensend/notifications/domain/DomainFactory.kt
package com.oxygensend.notifications.domain

import com.oxygensend.notifications.context.dto.MailDto
import com.oxygensend.notifications.context.dto.SmsDto

class DomainFactory {
    companion object {
        @JvmStatic
        fun from(mailDto: MailDto): Mail {
            return Mail(mailDto.subject, mailDto.body)
        }

        @JvmStatic
        fun from(smsDto: SmsDto): Sms {
            return Sms(smsDto.body)
        }

        @JvmStatic
        fun from(email: String?): Email {
            return Email(email!!)
        }

    }
}
